backend-api/src/functions/asyncActiveSession/tokenService/tokenService.ts

```diff
@@ -43,75 +56,74 @@ export class TokenService implements ITokenService {
       new Uint8Array(Buffer.from(encryptedCek, "base64")),
     );
     if (getCekResult.isError) {
-      return errorResult({
-        errorMessage: getCekResult.value.errorMessage,
-        errorCategory: getCekResult.value.errorCategory,
-      });
+      return getCekResult;
     }
 
     return successResult("");
   };
 
-  private readonly fetchPublicKey = async (
-    stsJwksEndpoint: string,
-  ): Promise<Result<JSON>> => {
-    let response;
-    try {
-      response = await fetch(stsJwksEndpoint, {
-        method: "GET",
-      });
-    } catch {
-      return errorResult({
-        errorMessage: "Unexpected error retrieving STS public key",
-        errorCategory: "SERVER_ERROR",
-      });
+  private readonly isJwks = (data: unknown): data is IJwks => {
+    return (
+      typeof data === "object" &&
+      data !== null &&
+      "keys" in data &&
+      Array.isArray((data as { keys: unknown }).keys) &&
+      (data as { keys: unknown[] }).keys.every(
+        (key) => typeof key === "object" && key !== null,
+      )
+    );
+  };
+
+  private async getCek(encryptedCek: Uint8Array): Promise<Result<Uint8Array>> {
+    const decryptCekResult = await this.kmsAdapter.decrypt(encryptedCek);
+    if (decryptCekResult.isError) {
+      return decryptCekResult;
     }
 
-    if (!response.ok) {
+    const cek = decryptCekResult.value.Plaintext ?? null;
+    if (cek === null) {
       return errorResult({
-        errorMessage: "Error retrieving STS public key",
+        errorMessage:
+          "No Plaintext received when calling KMS to decrypt the Content Encryption Key",
         errorCategory: "SERVER_ERROR",
       });
     }
 
-    let publicKey;
+    return successResult(cek);
+  }
+
+  private async getJwksFromResponse(
+    response: Response,
+  ): Promise<Result<IJwks>> {
+    let jwks;
     try {
-      publicKey = await response.json();
+      jwks = await response.json();
```

Comment:
If you return the response as string then here you could use `JSON.parse()`

Reply:
Updated